requests/models.py

```diff
@@ -154,8 +154,12 @@ def _encode_files(files, data):
 
             if isinstance(fp, (str, bytes, bytearray)):
                 fdata = fp
-            else:
+            elif hasattr(fp, 'read'):
```

Comment:
Sorry that I randomly come out of nowhere once in a blue moon to give review comments so ignore my comment if you like :) But hasattr has a small edge case which can cause an unexpected error in Python 2. You can refer to the link here:
https://hynek.me/articles/hasattr/

The article does mention that if you know what you are doing then its all good. My recommendation is to use getattr with default value of None to avoid AttributeError